File: FLM_SubconLabelSystem/Library/Library.Database/BLL/Log.cs
using System.Data;

namespace Library.Database.BLL
{
    /// <summary>
    /// Business Logic Layer
    /// ---------------------------------
    /// 18 Feb 2012   Yeon    Initial Version
    /// </summary>
    public class Log : Library.Root.Other.BusinessLogicBase
    {
        /// <summary>
        /// Retrieve the log (Common Function)
        /// Based on the Setup Key, the View's Name will be retrieved from the resource page
        /// Select the data based on the key, and the page no. was applied
        /// </summary>
        public static ListCollection GetLogList(string Table, string Key, int Page, string Sort = "")
        {
            ListCollection result = new ListCollection();

            if (Table != string.Empty)
            {
                using (var _dal = new DAL.Log())
                {
                    result = _dal.getLogList(Table, Key, FromRowNo(Page), ToRowNo(Page), Sort);
                }
            }
            return result;
        }

        public static DataTable GetGroupNo()
        {
            using (var _Dal = new DAL.Log())
            {
                return _Dal.GetGroupNo();
            }
        }
    }
}

File: FLM_SubconLabelSystem/Library/Library.Database/BLL/Log.vb
Namespace BLL
    ''' <summary>
    ''' Business Logic Layer
    ''' ---------------------------------
    ''' 18 Feb 2012   Yeon    Initial Version
    ''' </summary>
    ''' <remarks></remarks>
    Public Class Log
        Inherits Library.Root.Other.BusinessLogicBase
        ''' <summary>
        ''' Retrieve the log (Common Function)
        ''' Based on the Setup Key, the View's Name will be retrieved from the resource page
        ''' Select the data based on the key, and the page no. was applied
        ''' </summary>
        Public Shared Function GetLogList(ByVal Table As String, ByVal Key As String, ByVal Page As Integer, Optional ByVal Sort As String = "") As ListCollection
            GetLogList = New ListCollection

            If Table <> String.Empty Then
                Using _dal As New DAL.Log
                    GetLogList = _dal.getLogList(Table, Key, FromRowNo(Page), ToRowNo(Page), Sort)
                End Using
            End If
        End Function
        Public Shared Function GetGroupNo() As DataTable
            Using _Dal As New DAL.Log

                GetGroupNo = _Dal.GetGroupNo()

            End Using
        End Function

    End Class
End Namespace
